test/test_bit_vector.cpp

```diff
@@ -1,190 +1,199 @@
-    }
 
-    {
-        using pisa::broadword::msb;
-        std::vector<size_t> positions(1);
-        for(uint64_t i : ints) {
-            positions.push_back(positions.back() + msb(i) + 1);
-        }
+        {
+            using pisa::broadword::msb;
+            std::vector<size_t> positions(1);
+            for (uint64_t i : ints) {
+                positions.push_back(positions.back() + msb(i) + 1);
+            }
 
-        pisa::bit_vector_builder bvb(positions.back());
+            pisa::bit_vector_builder bvb(positions.back());
 
-        for (size_t i = 0; i < positions.size() - 1; ++i) {
-            uint64_t v = ints[i];
-            uint64_t len = positions[i + 1] - positions[i];
-            bvb.set_bits(positions[i], v, len);
-        }
+            for (size_t i = 0; i < positions.size() - 1; ++i) {
+                uint64_t v = ints[i];
+                uint64_t len = positions[i + 1] - positions[i];
+                bvb.set_bits(positions[i], v, len);
+            }
 
-        pisa::bit_vector bitmap(&bvb);
-        for (size_t i = 0; i < positions.size() - 1; ++i) {
-            uint64_t v = ints[i];
-            uint64_t len = positions[i + 1] - positions[i];
-            REQUIRE(v == bitmap.get_bits(positions[i], len));
+            pisa::bit_vector bitmap(&bvb);
+            for (size_t i = 0; i < positions.size() - 1; ++i) {
+                uint64_t v = ints[i];
+                uint64_t len = positions[i + 1] - positions[i];
+                REQUIRE(v == bitmap.get_bits(positions[i], len));
+            }
         }
-    }
+    });
 }
 
 TEST_CASE("bit_vector_enumerator")
 {
-    srand(42);
-    std::vector<bool> v = random_bit_vector();
-    pisa::bit_vector bitmap(v);
+    rc::check([](std::vector<bool> v) {
+        pisa::bit_vector bitmap(v);
 
-    size_t i = 0;
-    size_t pos = 0;
+        size_t i = 0;
+        size_t pos = 0;
 
-    pisa::bit_vector::enumerator e(bitmap, pos);
-    while (pos < bitmap.size()) {
-        bool next = e.next();
-        MY_REQUIRE_EQUAL(next, v[pos], "pos = " << pos << " i = " << i);
-        pos += 1;
+        pisa::bit_vector::enumerator e(bitmap, pos);
+        while (pos < bitmap.size()) {
+            bool next = e.next();
+            MY_REQUIRE_EQUAL(next, v[pos], "pos = " << pos << " i = " << i);
+            pos += 1;
 
-        pos += size_t(rand()) % (bitmap.size() - pos + 1);
-        e = pisa::bit_vector::enumerator(bitmap, pos);
-        i += 1;
-    }
+            pos += size_t(rand()) % (bitmap.size() - pos + 1);
+            e = pisa::bit_vector::enumerator(bitmap, pos);
+            i += 1;
+        }
+    });
 }
 
 TEST_CASE("bit_vector_unary_enumerator")
 {
-    srand(42);
-    uint64_t n = 20000;
-    std::vector<bool> v = random_bit_vector(n);
-
-    // punch some long gaps in v
-    for (size_t g = 0; g < n / 1000; ++g) {
-        ssize_t l = std::min(ssize_t(rand() % 256), ssize_t(v.size() - g));
-        std::fill(v.begin(), v.begin() + l, 0);
-    }
```

Comment:
@amallia Do you have any idea what actually was the point of this? It's weird, because it always fills from the beginning of the vector to some `l`, so effectively it will just make one "gap" at the beginning, won't it?

Reply:
It looks like a bug actually. Nice catch!

Reply:
So then, the intention is to introduce some contiguous spans of zeroes in the vector, is that correct?